guides/theming-your-components.md

```diff
@@ -11,46 +11,152 @@ customize components. Angular Material provides APIs for reading values from thi
-returns a Sass map containing the theme's primary, accent, and warn palettes, as well as a flag
-indicating whether dark mode is set.
+To read color values from a theme, you can use the `get-theme-color` Sass function. This function
+supports reading colors for both the app color palettes (primary, accent, and warn), as well as the
+foreground and background palettes. `get-theme-color` takes three arguments: The theme to read from,
+the name of the palette, and the name of the color.
+
+Each of the color palettes (primary, accent, and warn) supports reading the following named colors:
+
+| Color Name       | Description                                                                                                                                                        |
+|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| default          | The default color from this palette                                                                                                                                |
+| lighter          | A lighter version of the color for this palette                                                                                                                    |
+| darker           | A darker version of the color for this palette                                                                                                                     |
+| text             | The text color for this palette                                                                                                                                    |
+| default-contrast | A color that stands out against the this palette's default color                                                                                                   |
+| lighter-contrast | A color that stands out against the this palette's lighter color                                                                                                   |
+| darker-contrast  | A color that stands out against the this palette's darker color                                                                                                    |
+| [hue]            | The [hue] color for the palette.<br />[hue] can be one of: 50, 100, 200, 300, 400, 500, 600, 700, 800, 900, A100, A200, A400, A700                                 |
+| [hue]-contrast   | A color that stands out against the [hue] color for the palette.<br />[hue] can be one of: 50, 100, 200, 300, 400, 500, 600, 700, 800, 900, A100, A200, A400, A700 |
+
+The background palette supports reading the following named colors:
+
+| Color Name               | Description                                        |
+|--------------------------|----------------------------------------------------|
+| status-bar               | The background color for a status bar              |
```

Comment:
Which of these should I actually document? It's clearly just become a bag of "whatever I needed to implement X component"

Reply:
It's not really clear to me why we ever got so specific with these. Seems like we ought to just show them all though, since there isn't really a clear dividing line